receiver/httpcheckreceiver/scraper.go

```diff
@@ -122,6 +183,30 @@ func (h *httpcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error)
 			}
 
 			mux.Lock()
+
+			// Check if this is an HTTPS endpoint and extract TLS certificate info
+			// Default to true if CollectTLS is nil
+			collectTLS := h.cfg.Targets[targetIndex].CollectTLS == nil || *h.cfg.Targets[targetIndex].CollectTLS
+			if collectTLS {
+				parsedURL, urlErr := url.Parse(h.cfg.Targets[targetIndex].Endpoint)
+				if urlErr == nil && parsedURL.Scheme == "https" {
+					// Get TLS state by making a direct TLS connection
+					tlsState, tlsErr := getTLSState(h.cfg.Targets[targetIndex].Endpoint)
+					if tlsErr == nil && tlsState != nil {
+						issuer, commonName, sans, timeLeft := extractTLSInfo(tlsState)
```

Comment:
instead of making a new request here, would it be possible to use the TLS state from the response above (at line 172)? https://pkg.go.dev/net/http#Response

Reply:
Good call. I got it down to one request and it still has the same outcome.